pallets/micropayment/src/tests.rs

```diff
@@ -322,6 +323,10 @@ fn signature() {
 
     let pk = Public::from_raw(pk);
     let sig = Signature::from_slice(&sig);
-    let verified = sr25519_verify(&sig, &msg, &pk);
```

Comment:
![image](https://user-images.githubusercontent.com/4054836/162120302-46506952-71b0-4b1b-828f-7cda09c63ab9.png)

Since the `from_slice` method already done `copy_from_slice`, I think it's no need to do `copy_from_slice` again, 

```rust
let mut sig = [0u8; 64];
sig.copy_from_slice(&signature);
```
These 2 lines can be removed for simplicity

Reply:
there is a error in cargo test -p "" after i rm "sig.copy_from_slice(&signature);" in pub fn verify_signature
```
cargo test -p pallet-micropayment
   Compiling pallet-micropayment v3.0.0 (/Users/li/Documents/GitHub/deeper-chain/pallets/micropayment)
warning: unused variable: `signature`
   --> pallets/micropayment/src/lib.rs:495:13
    |
495 |             signature: &Vec<u8>,
    |             ^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_signature`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: variable does not need to be mutable
   --> pallets/micropayment/src/lib.rs:501:17
    |
501 |             let mut sig = [0u8; 64];
    |                 ----^^^
    |                 |
    |                 help: remove this `mut`
    |
    = note: `#[warn(unused_mut)]` on by default

warning: `pallet-micropayment` (lib) generated 2 warnings
warning: `pallet-micropayment` (lib test) generated 2 warnings (2 duplicates)
    Finished test [unoptimized + debuginfo] target(s) in 3.21s
     Running unittests (target/debug/deps/pallet_micropayment-479c432e94837d39)

running 9 tests
test tests::signature ... ok
test mock::__construct_runtime_integrity_test::runtime_integrity_tests ... ok
test tests::test_blake2_hash ... ok
test tests::claim_payment ... FAILED
test tests::open_channel ... ok
test benchmarking::tests::test_benchmarks ... ok
test tests::add_balance ... ok
test tests::close_expired_channels ... ok
test tests::close_channel ... ok

failures:

---- tests::claim_payment stdout ----
[
    0xf4,
    0x3d,
    0x4f,
    0x87,
    0xe8,
    0x6e,
    0x99,
    0xba,
    0x1e,
    0x82,
    0x49,
    0x7a,
    0xf,
    0x5d,
    0x54,
    0xc5,
    0x27,
    0xb5,
    0xe2,
    0xe,
    0x2c,
    0x50,
    0xa7,
    0x1c,
    0xc,
    0xde,
    0xa5,
    0xdc,
    0xe,
    0x63,
    0x14,
    0xc5,
]
thread 'tests::claim_payment' panicked at 'Expected Ok(_). Got Err(
    DispatchErrorWithPostInfo {
        post_info: PostDispatchInfo {
            actual_weight: None,
            pays_fee: Pays::Yes,
        },
        error: Module(
            ModuleError {
                index: 4,
                error: 6,
                message: Some(
                    "InvalidSignature",
                ),
            },
        ),
    },
)', pallets/micropayment/src/tests.rs:273:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    tests::claim_payment

test result: FAILED. 8 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.40s

error: test failed, to rerun pass '-p pallet-micropayment --lib'
```

Reply:
Sorry I'm pointing on the wrong line. I'm the 
`from_slice` method includes `r.copy_from_slice(data);` so we don't need to call `r.copy_from_slice(data);` in micropayment/src/lib.rs again.